src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,77 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Collections;
+
 public class WorkWithFile {
+    static final int INDEX_ZERO = 0;
+    static final int INDEX_ONE = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        ArrayList<String> strings = new ArrayList<>();
+        readFile(strings, fromFileName);
+        generateReport(strings);
+        writeToFile(strings, toFileName);
```

Comment:
```suggestion
        String dataFromFile = readFile(fromFileName);
        String report = generateReport(dataFromFile);
        writeToFile(dataFromFile, toFileName);
```
if you want to pass the result of the first method to the second - make the first method return the result instead of using some common object to share the result. refactor your code so main method will look like above